app/models/concerns/push_updates_for_subscription.rb

```diff
@@ -39,4 +29,26 @@ def app_name
     end
   end
 
+  def data_to_send
+    data = { category: 'message',
+             message: message_body,
+             is_private: message.is_private,
+             item_id: item_id,
+             author_id: message.sender_id,
+             message_id: message.id }
+    data[identity] = message.related_object.id
+    data
+  end
+
+  def message_body
+    message.body.truncate(150, separator: ' ')
+  end
+
+  def item_id
+    message.messageable.instance_of?(Item) && message.messageable_id
+  end
+
+  def identity
```

Comment:
as commented above, the apps need to start to understand `messageable_type` and `messageable_id` . Don't include a new method called `identity`

Reply:
understood this is for backwards compatibility